getstation/theme
# [APP-645] Use generated SVG components for icons rather than loading a SVG file

Objective: no dependency on a static SVG file

### Code changes summary
- Cut the `Icons.svg` into smaller independent SVG files (using [this](https://gist.github.com/alexstrat/d5ce0a2f23e4f549ca65f3bc47483b0f)) 
- Implemented `generate:icons` script that'll generate `SVGComponents` using [svgr](https://github.com/smooth-code/svgr) and a file with enums (no braking change compared to today)
- Use the generated SVG in Icon

### Breaking change for user of library
It is not necessary anymore to import the file `static/Icons.svg` in your project, as `Icon` component already contains SVG source.

Moreover the file `static/Icons.svg`  **does not exist anymore**.

### Test instructions
- launch storybook
- verify the diff in the `All Icons` story with the [one online](https://getstation.github.io/theme/storybook/index.html?knob-value=Trello&knob-label=Email&knob-Error=invalid%20value&selectedKind=Atoms%7CIcon&selectedStory=All%20icons&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Fnotes%2Fpanel)
- => no diff except:
  - empty icons have disappeared
  - the `app-icon-placeholder` has changed a bit => I will verify in appstroe it does not break anything.

### Further possible improvements
- Make sure every icon has consistent and minimal padding (see example of guidelines [here](https://material.io/design/iconography/system-icons.html#grid-keyline-shapes)) 
- generate with [icon option](https://www.smooth-code.com/open-source/svgr/docs/options/#icon) so that icons have by default line height
- get rid of the `Symbol` thing
